### ISSUE TITLE
Setting a property counts as adding an attribute (too-many-instance-attributes)

### ISSUE THREAD

**Issue by ManelBH** [PyCQA/pylint#4100]
### Steps to reproduce
1. Create file `test.py`

```python
# pylint: disable=missing-module-docstring, missing-class-docstring, missing-function-docstring
# pylint: disable=too-few-public-methods, blacklisted-name,unused-private-member


class Foo:
    def __init__(self):
        self.__one = None
        self.__two = None
        self.__three = None
        self.__four = None
        self.__five = None
        self.__six = None
        self.__seven = None

    @property
    def one(self):
        return self.__one

    @one.setter
    def one(self, value):
        self.__one = value


class Bar:
    def __init__(self):
        self.__one = None
        self.__two = None
        self.__three = None
        self.__four = None
        self.__five = None
        self.__six = None
        self.__seven = None

    @property
    def one(self):
        return self.__one

    @one.setter
    def one(self, value):
        self.__one = value


def test():
    bar = Bar()
    bar.one = 1

```
2. Run `python -m pylint test.py`

### Current behavior
pylint outputs:
`test.py:22:0: R0902: Too many instance attributes (8/7) (too-many-instance-attributes)`
(pylint believes Bar has 8 attributes, but not Foo)

### Expected behavior
1. Both classes should produce the same result
2. Setting a property shouldn't count as adding an attribute.

### pylint --version output
pylint 2.5.2
astroid 2.4.1
Python 3.8.5 (tags/v3.8.5:580fbb0, Jul 20 2020, 15:57:54) [MSC v.1924 64 bit (AMD64)]

**Comment by Pierre-Sassoulas** [comment]
Thank you for the issue, I can reproduce this on the latest master.

**Comment by Pierre-Sassoulas** [comment]
Using 
```python
# pylint: disable=missing-module-docstring, missing-class-docstring, missing-function-docstring
# pylint: disable=too-few-public-methods, blacklisted-name,unused-private-member


class Foo:
    def __init__(self):
        self.one = None
        self.two = None

    @property
    def one(self):
        return self.__one

    @one.setter
    def one(self, value):
        self.__one = value


class Bar:
    def __init__(self):
        self.__one = None
        self.__two = None

    @property
    def one(self):
        return self.__one

    @one.setter
    def one(self, value):
        self.__one = value


def test():
    bar = Bar()
    bar.one = 1
```
And launching ``pylint a.py --enable=too-many-instance-attributes --max-attributes=2`` now has the same result for both classes:
```
a.py:5:0: R0902: Too many instance attributes (3/2) (too-many-instance-attributes)
a.py:19:0: R0902: Too many instance attributes (3/2) (too-many-instance-attributes)
```

The issue about the property / attribute that should be counted the same. still exists. But this is a simple case where we create a getter/setter when we could do without (by just making one public), so we could potentially detect that it's the same attribute. But in the general case each attribute / property makes the class more complex and it would also be harder to detect. Ie does this have 4 attributes or 2 ? I'd say it's 4:

```python
import math


class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    @property
    def r(self):
        return math.sqrt(self.x * self.x + self.y * self.y)

    @property
    def phi(self):
        return math.tanh(self.y / self.x)
```
